observability/opentelemetry-api/src/main/java/org/wildfly/extension/opentelemetry/api/WildFlyOpenTelemetryConfig.java

```diff
@@ -101,13 +49,114 @@ public boolean isMpTelemetryInstalled() {
         return mpTelemetryInstalled;
     }
 
-    /**
-     *  Only add the value to the config if it is non-null, and convert the type to String to
-     *  satisfy library requirements.
-     */
-    private void addValue(Map<String, String> config, String key, Object value) {
-        if (value != null) {
-            config.put(key, value.toString());
+    public static class Builder {
+        final Map<String, String> properties = new HashMap<>();
+        private boolean mpTelemetryInstalled;
+
+        public Builder() {
+            addValue(OTEL_EXPORTER_OTLP_PROTOCOL, "grpc");
+            addValue(OTEL_PROPAGATORS, "tracecontext,baggage");
+            addValue(OTEL_SDK_DISABLED, "false");
+
+        }
+
+        public Builder setServiceName(String serviceName) {
+            addValue(OTEL_SERVICE_NAME, serviceName);
+            return this;
+        }
+
+        public Builder setSdkDisabled(boolean sdkDisabled) {
+            addValue(OTEL_SDK_DISABLED, "false");
+            return this;
+        }
```

Comment:
I won't hold merging for this as it's unused, but please file a JIRA to get this removed.

Reply:
I added that as I thought we'd need that in the MP module, but I forgot that, there, we get the props map then override as needed. JIRA: https://issues.redhat.com/browse/WFLY-20158